src/api/CheckListItemApi.ts

```diff
@@ -155,10 +155,17 @@ export const useOpenChecklistItems = () => {
 /**
  * Gets all checklist items for the current user
  */
-export const useMyChecklistItems = (user: Person, roles: RoleType[]) => {
+export const useMyChecklistItems = (
+  user: Person,
+  roles: RoleType[],
+  fetchCompleted: boolean
+) => {
   return useQuery({
-    queryKey: ["myChecklist", user, roles],
-    queryFn: () => getMyCheckListItems(user, roles),
+    queryKey: fetchCompleted ? ["myChecklist", user, roles] : ["checklist"],
+    queryFn: () =>
+      fetchCompleted
+        ? getMyCheckListItems(user, roles)
+        : getOpenCheckListItems(),
```

Comment:
Can you take the combining one step further, and just pass the boolean on whether or not to get completed items to the function called by the hook too?  Then just add the filter for Completed or not based on that boolean.  I like the filtering for Lead you are doing server side, so might as well leverage that for both queries.

Reply:
So, this is where we hope they gave us good info.  I believe they have numbers less than that per year for In/Out processing -- And, they were looking at purging every quarter, so, if giving some grace period, 3 times a year, they should be under that every 4 months.